src/gluonts/dataset/util.py

```diff
@@ -73,9 +74,7 @@ def get_bounds_for_mp_data_loading(dataset_len: int) -> DataLoadingBounds:
     return DataLoadingBounds(lower=lower, upper=upper)
 
 
-def _split(
-    it: Iterator[T], fn: Callable[[T], bool]
-) -> Tuple[List[T], List[T]]:
+def _split(it: List[T], fn: Callable[[T], bool]) -> Tuple[List[T], List[T]]:
```

Comment:
What's the reason for this change?

Reply:
I was using lists for some operations and it was simpler to just use them everywhere instead of passing `iter(mylist)` to please mypy. Also materializing the list of file names is never a problem.